core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1910,6 +1910,16 @@ package object config {
       .booleanConf
       .createWithDefault(true)
 
+  private[spark] val IO_COMPRESSION_ZSTD_WORKERS =
+    ConfigBuilder("spark.io.compression.zstd.workers")
+      .doc("Thread size spawned to compress in parallel when using Zstd. When value <= 0, " +
+        "no worker is spawned, it works in single-threaded mode. When value > 0, it triggers " +
+        "asynchronous mode, corresponding number of threads are spawned. More workers improve " +
+        "performance, but also increase memory cost.")
+      .version("4.0.0")
+      .intConf
+      .createWithDefault(8)
```

Comment:
This does help for those IO-heavy workloads but uses less CPU, but I think we should use the single thread in default, otherwise, it would make task shuffle writing occupy additional CPU(threads), IMO it does not fit the Spark executor's thread-based parallel computing model.

Reply:
Ya, we should use `0` by default because this has a side-effect on both CPU cycle and native Memory usage. The production job can have perf regressions.

Reply:
> IMO it does not fit the Spark executor's thread-based parallel computing model.

I can't entirely agree with your opinion on confusing Spark executors' cores with threads. But I'm okay with 0 as the default.